compiler/rustc_mir/src/transform/const_prop.rs

```diff
@@ -738,18 +785,24 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
         rvalue: &Rvalue<'tcx>,
         place: Place<'tcx>,
     ) -> Option<()> {
+        let tcx = self.tcx;
         self.use_ecx(|this| {
             match rvalue {
                 Rvalue::BinaryOp(op, box (left, right))
                 | Rvalue::CheckedBinaryOp(op, box (left, right)) => {
                     let l = this.ecx.eval_operand(left, None);
                     let r = this.ecx.eval_operand(right, None);
+                    Self::check_interpresult(tcx, &l);
+                    Self::check_interpresult(tcx, &r);
```

Comment:
Oh wow this is so fragile... I can guarantee that future ConstProp edits will forget to add this somewhere. We need to find a better solution than this.

Reply:
Also, I don't think making ConstProp so much more complicated just for the sake of slightly better diagnostics on compiler OOM is justified. This entire PR is a diagnostic fix, right? Compilation is going to fail anyway due to this OOM. If we make soundness-critical code such as ConstProp more complicated and that introduces a bug down the road, I think that's worse than our current ICEs on OOM.

See https://github.com/rust-lang/rust/pull/86255#issuecomment-871601495 for a proposal that hopefully will require barely any changes in ConstProp.